persistence/src/commonMain/db_user/com/wire/kalium/persistence/Messages.sq

```diff
@@ -518,3 +518,10 @@ WHERE text LIKE ('%' || :searchQuery || '%')
 AND conversationId = :conversationId
 AND contentType = 'TEXT'
 ORDER BY date DESC;
+
+selectSearchedConversationMessagePosition:
+SELECT COUNT(*) FROM Message
+WHERE
+conversation_id = :conversationId
+AND creation_date >= (SELECT creation_date FROM Message WHERE id = :messageId LIMIT 1)
```

Comment:
If ordered directly through the `id` we won't be getting the correct messages as `messageId` is created using `uuid4` (uuid v4) and not `uuid v1` (which generates the id based on the time as well).

Reply:
we keep the order with date but query by ID

Reply:
Oh sorry I expressed myself wrong, instead of ordering I meant querying, so if we have like 100 messages and we want message on position 90:

if done with `creation_date >= ...` it returns correctly 90
if done with just the id it doesn't return the correct position (rather less value than the correct one)

also, if we just query by id as we are using `COUNT` it will return just 1 always as we will be doing the query WHERE conversation ID = x and message ID = y, there is no count 🤔